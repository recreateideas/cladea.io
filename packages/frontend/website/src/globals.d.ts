import { Theme } from "./App/ui-core/theme";
import { Store } from "redux";

export declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: any;
    /** enzyme global methods for testing */
    shallow: any;
    render: any;
    mount: any;
    theme: Theme;
    ResizeObserver: any;
    store: Store;
  }
}

declare module "ui-core";
