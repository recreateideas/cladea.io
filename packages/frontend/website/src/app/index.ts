export * from "./App";
export * from "./features";
