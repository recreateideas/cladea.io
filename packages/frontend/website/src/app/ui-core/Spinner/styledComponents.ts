import styled, { css } from 'styled-components';

export const Container = styled.div<{
    speed?: 'slow' | 'normal' | 'fast' | 'superFast';
}>`
    ${(props) => {
        const {
            theme: {
                dsl: { styles },
            },
            speed = 'fast',
        } = props;
        const animationSpeed = styles.animationSpeed[speed];
        return css`
            span {
                animation-duration: ${animationSpeed};
            }
        `;
    }}
`;
