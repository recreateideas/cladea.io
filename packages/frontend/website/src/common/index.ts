export * from './utils';
export * from './hooks';
export * from './converters';
