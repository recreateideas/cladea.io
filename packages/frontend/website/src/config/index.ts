export const apiGatewaUrl = process.env.REACT_APP_API_GATEWAY_URL;

console.log(process.env);
