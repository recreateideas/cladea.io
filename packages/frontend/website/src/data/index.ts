import land110m from "./land-110m.json";

export { land110m };
