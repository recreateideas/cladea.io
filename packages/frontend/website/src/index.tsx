import "./app/ui-core/theme/GlobalStyle/roboto.css";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { store } from "./redux";
import { ServicedApp, ThemeProviderConnected } from "./app";

ReactDOM.render(
  <Provider store={store}>
    <ThemeProviderConnected>
      <ServicedApp />
    </ThemeProviderConnected>
  </Provider>,
  document.getElementById("root")
);
