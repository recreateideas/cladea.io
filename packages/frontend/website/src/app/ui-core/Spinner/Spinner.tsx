import { useTheme } from 'styled-components';
import { ReactElement } from 'react';
import { CircularProgress, CircularProgressProps } from '@mui/material/';
import { Box } from '@mui/material';
import { Container } from './styledComponents';
import { get } from 'lodash';

interface IProps extends Omit<CircularProgressProps, 'color'> {
    withRail?: boolean;
    color?: string;
    error?: boolean;
    speed?: 'slow' | 'normal' | 'fast' | 'superFast';
}
export const Spinner = (props: IProps): ReactElement => {
    const {
        withRail = true,
        color: colorPath = 'secondary.yellow[900]',
        speed,
        disableShrink,
        error,
        ...progressProps
    } = props;
    const {
        //@ts-ignore
        dsl: { palette },
    } = useTheme();
    const color = get(palette, colorPath);
    return (
        <Container className="spinner" {...{ speed }}>
            <Box sx={{ display: 'flex' }}>
                {withRail && !error && (
                    <CircularProgress
                        sx={{
                            position: 'absolute',
                            color: () => palette.tertiary.grey[400],
                        }}
                        {...progressProps}
                        variant="determinate"
                        value={100}
                    />
                )}
                {error ? (
                    <CircularProgress
                        sx={{
                            color: () => palette.secondary.red[900],
                        }}
                        {...progressProps}
                        variant="determinate"
                        value={100}
                    />
                ) : (
                    <CircularProgress
                        {...progressProps}
                        disableShrink={disableShrink}
                        sx={{ color: () => color }}
                    />
                )}
            </Box>
        </Container>
    );
};

Spinner.displayName = 'Spinner';
