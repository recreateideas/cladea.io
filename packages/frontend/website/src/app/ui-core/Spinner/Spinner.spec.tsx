import { Spinner } from './Spinner';
import { TestProvider } from 'src/tests';

const { mountWithProviderAndTheme } = TestProvider();
describe('<Spinner/>', () => {
    it('should render', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Spinner />, state);
        expect(root).toMatchSnapshot();
    });
    it('should render the error more', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Spinner error />, state);
        expect(root).toMatchSnapshot();
    });
});
