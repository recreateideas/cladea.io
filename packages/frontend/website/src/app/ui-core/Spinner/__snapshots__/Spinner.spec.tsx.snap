// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner/> should render 1`] = `
Array [
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: inline-block;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #12D8F7;
  position: absolute;
  color: #8F8F8F;
}

.c4 {
  display: inline-block;
  color: #12D8F7;
  -webkit-animation: epmXAj 1.4s linear infinite;
  animation: epmXAj 1.4s linear infinite;
  color: #FDC01F;
}

.c2 {
  display: block;
}

.c3 {
  stroke: currentColor;
  -webkit-transition: stroke-dashoffset 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: stroke-dashoffset 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c5 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: jzfMP 1.4s ease-in-out infinite;
  animation: jzfMP 1.4s ease-in-out infinite;
}

<div
    className="spinner"
  >
    <div
      className="c0 MuiBox-root"
    >
      <span
        aria-valuenow={100}
        className="c1 MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "transform": "rotate(-90deg)",
            "width": 40,
          }
        }
      >
        <svg
          className="c2 MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="c3 MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={
              Object {
                "strokeDasharray": "126.920",
                "strokeDashoffset": "0.000px",
              }
            }
          />
        </svg>
      </span>
      <span
        className="c4 MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="c2 MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="c5 MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
  </div>,
  "",
]
`;

exports[`<Spinner/> should render the error more 1`] = `
Array [
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: inline-block;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #12D8F7;
  color: #F50933;
}

.c2 {
  display: block;
}

.c3 {
  stroke: currentColor;
  -webkit-transition: stroke-dashoffset 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: stroke-dashoffset 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

<div
    className="spinner"
  >
    <div
      className="c0 MuiBox-root"
    >
      <span
        aria-valuenow={100}
        className="c1 MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "transform": "rotate(-90deg)",
            "width": 40,
          }
        }
      >
        <svg
          className="c2 MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="c3 MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={
              Object {
                "strokeDasharray": "126.920",
                "strokeDashoffset": "0.000px",
              }
            }
          />
        </svg>
      </span>
    </div>
  </div>,
  "",
]
`;
